### ISSUE TITLE
Mapping turn-restrictions for navigation in 30 cities in US

### ISSUE THREAD

**Issue by srividyacb** [mapbox/mapping#187]
### Aim:
To map [turn-restrictions](http://wiki.openstreetmap.org/wiki/Relation:restriction) in 39 US cities using Mapillary as the source. 

#### Task Workflow

- **[Here](https://github.com/mapbox/mapping/wiki/Mapping-guide-for-adding-turn-restrictions-using-Mapillary)** is the workflow to follow to add turn-restrictions to OpenStreetMap.
- Here is the **[spreadsheet](https://docs.google.com/a/mapbox.com/spreadsheets/d/1OlhPq6j7tQiONrl4Iwq7ZoDHVJto_UZM_vWcDyBZ0d8/edit?usp=sharing)** where we will be tracking the progress.

------

![image](https://cloud.githubusercontent.com/assets/3423533/15504805/5d678d70-21de-11e6-8290-d20db2380d4d.png)

**Team** : @mapbox/team-data 

**Features to map** : Turn-restrictions by using mapillary as the main source and Bing/Mapbox Satellite imagery.

**Process:** 
- Two team members can pick one city together and update the spreadsheet with their name. 
- Before starting the mapping run the below overpass query by changing the city name in the query and update the spreadsheet with number of turn-restriction present before mapping. 
- **Overpass Query:** http://overpass-turbo.eu/s/gqr
- Divide the city into half based on a road/landmark and work at different parts of the city.

#### JOSM Setup

**Plugins:**

- [Mapillary plugin](http://blog.mapillary.com/update/2015/06/25/josm-mapillary.html)
- [Turn-Restriction plugin](http://wiki.openstreetmap.org/wiki/JOSM/Plugins/Turnrestrictions)

**Paint Styles**

- [Traffic Signs](https://raw.githubusercontent.com/species/josm-preset-traffic_sign_direction/master/direction.mapcss) : JOSM Preferences -> Map Settings -> traffic signs

**Changeset Comment:** `Updating turn restrictions in CITYNAME using Mapillary imagery https://github.com/mapbox/mapping/issues/187`

**Source:** Bing/Mapillary

**Comment by chtnha** [comment]
I came across few Mapillary images in Los Angeles where the turn-restriction was written on a turn-lane boards (white background and black signs). 

![screen shot 2016-05-25 at 3 48 33 pm](https://cloud.githubusercontent.com/assets/6770741/15566835/b32da380-2340-11e6-8776-2b7fa8427c4a.png)

![screen shot 2016-05-25 at 5 52 58 pm](https://cloud.githubusercontent.com/assets/6770741/15566807/9136e9e4-2340-11e6-9e4a-343f0177fde9.png)

In such situations make sure to add `no_u_turn` turn restrictions.

**Comment by jinalfoflia** [comment]
@mapbox/team-data have been working on adding turn-restrictions in 30 US Cities to improve the navigation data.
​
##### Process we are following

- Dividing a city to 2 or 3 people based on Mapillary coverge in that city.
- Update the number of turn-restriction already present in OpenStreetMap.
- Use [Mapillary traffic sign detection](http://planemad.github.io/osm-navigation/#19.2/36.10382/-115.17288/-0.9) to identify the turn-restriction present in the junction. 
- Click on the detected turn-sign to open in JOSM and follow the [workflow](https://github.com/mapbox/mapping/wiki/Mapping-guide-for-adding-turn-restrictions-using-Mapillary) to add data to OpenStreetMap.
- Once done with the city add ✔️  to the city in the table below and run the Overpass query to update the number of turn-restrictions added.
​

The number of turn restrictions already present in the cities listed below can be found [here](https://github.com/mapbox/mapping/blob/30cities/30cities/30cities_baseline_stats.csv). Below is the list of cities where we are adding turn-restrictions:

|Serial  No. | City | Team | No. of Restrictions Added | Status |
| --------- | ------- | ------- | --------- | --------- |
| 1 | Detroit  | @ramyaragupathy  | **0** | ✔️  |
| 2 | Los Angeles |  @ramyaragupathy @nikhilprabhakar @chtnha | **69** | ✔️   |
| 3 | Minneapolis and St Paul | @maning @krishnanammala  | **68** | ✔️  |
| 4 | New York City | @oini @jothirnadh | |**WIP** |
| 5 | Phoenix  | @planemad @poornibadrinath | | **WIP** |
| 6 | Boston | @srividyacb @pratikyadav | | **WIP** |
| 7 | Washington D C  | @jinalfoflia @abhisheksaikia | | **WIP** |
| 8 | Chicago | @Andygol | | **WIP** |
| 9 | Las Vegas | @srividyacb | | **WIP** |
|10 | Seattle | | | |
| 11 | Portland | | | |
| 12 | El Paso  | | | |
| 13 | Austin | | | |
| 14 | Huston  | | | |
| 15 | San Antonio | | | |
| 16 | Dallas | | | |

**Comment by Andygol** [comment]
Chicago is Done 🎯 
/cc @planemad

**Comment by planemad** [comment]
We have updates to the experimental OSM navigation map to help review and add turn restrictions from Mapillary quickly.

![untitled2](https://cloud.githubusercontent.com/assets/126868/15677497/23779148-2768-11e6-8dc0-1923abcc760e.gif)

**http://planemad.github.io/osm-navigation-map/** (Will be moved after testing)

**Features**
- Inspect existing turn restrictions
- Inspect Mapillary photos with detected turn restriction signages
- Add a reviewed marker with the status of the detected restriction:
 - `Valid` Restriction is correct and has been mapped
 - `Redundant` The restriction is correct but is not useful to add to the map. eg. No turn against a oneway.
 - `Invalid` The restriction was wrongly detected, is temporary or in the wrong location. Do not map.

Please submit feedback and file issues at https://github.com/planemad/osm-navigation-map/issues

cc @mapbox/team-data

**Comment by planemad** [comment]
Reminder to @mapbox/team-data to take care of **conditional restrictions** like this:

<img width="783" alt="screenshot 2016-06-01 17 52 07" src="https://cloud.githubusercontent.com/assets/126868/15709310/a6cfecb6-2821-11e6-8d25-e5001f638661.png">

The correct tagging is `restriction:conditional=no_left_turn @ (Mo-Fr 06:00-09:00,16:00-18:00)`. This is already documented in the workflow.

**Comment by poornibadrinath** [comment]
While marking turn restrictions near one ways as redundant, it would be great to check instances like this with u turns. 
![image](https://cloud.githubusercontent.com/assets/17887418/15742010/15c3aea6-28da-11e6-90f3-eba83f439499.png)
![image](https://cloud.githubusercontent.com/assets/17887418/15742028/2ed80e3c-28da-11e6-96e4-d5f1b364de4b.png)
While left and right turn restrictions near u turns are redundant, u turns on one ways can be valid. 

/cc: @mapbox/team-data

**Comment by Andygol** [comment]
@mapbox/team-data I want to attract your attention to this case of turn restriction https://www.openstreetmap.org/relation/2573759.

Here is `no_left_turn` restriction with 3 `via` members!

![image](https://cloud.githubusercontent.com/assets/369696/15681814/7bd9664a-2763-11e6-85b7-c09b67a208ea.png)

This restriction makes sense because of the markings before the crossroads. Here is solid marking so drivers not allowed to change their lanes from the node where `motorway_link` join to the `primary` road down to the intersection. Drivers traveling from the motorway may move only by the most right lane to right. On the other hand, drivers moving along `primary` _not allowed_ to make **right turn**, that described by this turn restriction - https://www.openstreetmap.org/relation/2573760

Please be careful when you are editing

**Comment by abhisheksaikia** [comment]
The Mapillary layer in the [navigation map](http://planemad.github.io/osm-navigation-map/#18.75/38.93635/-77.02470) is showing a series of possible turn restriction images. 

![screen shot 2016-06-03 at 3 37 20 pm](https://cloud.githubusercontent.com/assets/13744156/15775713/1b94b0f6-29a1-11e6-93e0-933e89815afa.png)

*A series of mapillary images with possible turn restrictions in Georgia Avenue*


But when I am opening the at same area in JOSM, the mapillary plugin is not displaying the series of images that are displayed in a line around "Georgia Avenue".

![screen shot 2016-06-03 at 3 40 18 pm](https://cloud.githubusercontent.com/assets/13744156/15775778/7f3f5a98-29a1-11e6-9572-3a2ca24708d2.png)

*No mapillary images displayed in JOSM*

The plugin has been updated and I have tried deleting and loading the mapillary layer again multiple times. The problem still persists.

Is this because the mapillary layer has not been updated?

cc @gyllen @planemad @jinalfoflia

**Comment by samely** [comment]
@planemad 
While working on turn restrictions, I found some stuff that I think could be improved.
- Could you change the style of points? In the tool, you easily see the red and the bright green ones, but we need to fix just the green ones. So, if you could change the color and size of the points that aren't already done, would be great.
![image](https://cloud.githubusercontent.com/assets/10425629/15791142/3b189d26-299b-11e6-9f97-a434c2286705.png)

- When you work in a certain city, we just see WIP, but we don't know what  part is already done. So I think in the spreadsheet we could add a column to add the covered area i.e. Done the north part of Ventura fway in NorthWest Los Angeles

**Comment by jinalfoflia** [comment]
### Week [ 30th May 2016 - 3rd June 2016]

The team has completed adding turn restrictions in a total of **12 cities** since 25th May 2016. This week the team completed **9 cities** and added **216** turn restrictions in the following cities:

- Austin
- Baltimore
- Boston
- Columbus
- Houston
- Indianapolis
- New york
- Pheonix
- Portland
- Seattle 
- DC
- Chicago

Please refer to the [above table](https://github.com/mapbox/mapping/issues/187#issuecomment-222290707) to get more details about the cities completed and the cities that we are currently working on.

/cc @planemad @maning

**Comment by planemad** [comment]
@samely looks like there was some confusion in the style. have updated it:
- Blue markers are the areas that have been reviewed
- Green circles are detected Mapillary signage (visible only after z14)

<img width="522" alt="openstreetmap_navigation_data_map" src="https://cloud.githubusercontent.com/assets/126868/15800211/b989c860-2a90-11e6-9f3a-29175afcedb3.png">

**Comment by jinalfoflia** [comment]
Great work team 👏 
A team of **14 members** worked on **32 US cities** adding **1133** turn restrictions in **13 days**. The time spent on each of the cities varies based on the Mapillary coverage and number of turn restrictions detected.  The city-wise stats can be found [here](https://github.com/mapbox/mapping/issues/187#issuecomment-222290707).💥 

### Next Action[s]

- The team will work on a work-flow to validate the added turn-restrictions.
- The team will work on re-reviewing all the cities based on the workflow, to ensure that all the detected turn-restrictions are mapped correctly. 

/cc @mapbox/team-data

**Comment by jinalfoflia** [comment]
> The team will work on a work-flow to validate the added turn-restrictions.

This is the trial run on the proposed work-flow to be followed to validate the turn restrictions added by the team in the 30 US cities.

**Team:** @abhisheksaikia @jinalfoflia @oini 

- Use [Overpass](http://overpass-turbo.eu/) to extract the turn restrictions added by the team from **25th June to 11 July**
- Create the [.osm](http://wiki.openstreetmap.org/wiki/OSM_file_formats) per city and share it with the team
- The team will validate the turn restrictions added in these cities.

**Changeset comment to be used:** `Validating turn restrictions in the US cities using Mapillary imagery https://github.com/mapbox/mapping/issues/187`

/cc @planemad @abhisheksaikia @oini

**Comment by jinalfoflia** [comment]
@mapbox/team-data team has completed the validation of close to **900 turn restrictions** mapped in the 30 US cities using the above work-flow. The team will next jump into validating **300 odd**  `conditional turn-restrictions` in these cities.

**Comment by jinalfoflia** [comment]
@mapbox/team-data completed validating the conditional turn restrictions in the 30 US cities.💥 No further actions here. Closing it.
